Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -86,4 +91,96 @@ namespace ROS2
         return joints;
     }
 
+    AZStd::unordered_set<AZStd::string> Utils::GetMeshesFilenames(const urdf::LinkConstSharedPtr& root_link, bool visual, bool colliders)
+    {
+        AZStd::unordered_set<AZStd::string> filenames;
+        const auto getFnFromGeometry = [](const urdf::GeometrySharedPtr& geometry)
+        {
+            if (geometry->type == urdf::Geometry::MESH)
+            {
+                auto pMesh = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+                if (!pMesh)
+                {
+                    return AZStd::string{};
+                }
+                return AZStd::string(pMesh->filename.c_str(), pMesh->filename.size());
+            }
+            return AZStd::string{};
+        };
+
+        const auto processLink = [&](const urdf::Link& link)
```

Comment:
this actually only reads two bools and appends to "filenames" but captures everything